rust/src/nfs/nfs.rs

```diff
@@ -1799,57 +1855,93 @@ pub extern "C" fn rs_nfs_probe_ms(
             } else {
                 SCLogDebug!("nfs_probe_dir said STREAM_TOCLIENT");
             }
-            let r = nfs_probe(slice, adirection);
-            if r == 1 {
-                SCLogDebug!("nfs_probe success: dir {:02x} adir {:02x}", direction, adirection);
-                if (direction & (STREAM_TOSERVER|STREAM_TOCLIENT)) != adirection {
-                    unsafe { *rdir = adirection; }
-                }
-                return 1;
+            match nfs_probe(slice, adirection) {
+                1 => {
+                    SCLogDebug!("nfs_probe success: dir {:02x} adir {:02x}", direction, adirection);
+                    if (direction & (STREAM_TOSERVER|STREAM_TOCLIENT)) != adirection {
+                        unsafe { *rdir = adirection; }
+                    }
+                    return unsafe { ALPROTO_NFS };
+                },
+                0 => { return ALPROTO_UNKNOWN },
+                _ => { return unsafe { ALPROTO_FAILED } },
             }
-            return r;
         },
         0 => {
-            return 0;
+            return ALPROTO_UNKNOWN;
         },
         _ => {
-            return -1;
+            return unsafe { ALPROTO_FAILED };
         }
     }
 }
 
 #[no_mangle]
-pub extern "C" fn rs_nfs_probe(direction: u8,
-        input: *const u8, len: u32)
-    -> i8
+pub extern "C" fn rs_nfs_probe(_f: *const Flow,
+                               direction: u8,
+                               input: *const u8,
+                               len: u32,
+                               _: *mut u8)
+    -> AppProto
 {
+    if len < NFS_MIN_FRAME_LEN as u32 {
+        return ALPROTO_UNKNOWN;
+    }
+
     let slice: &[u8] = build_slice!(input, len as usize);
     SCLogDebug!("rs_nfs_probe: running probe");
-    return nfs_probe(slice, direction);
+    match nfs_probe(slice, direction) {
+        1 => { return unsafe { ALPROTO_NFS } },
+        -1 => { return unsafe { ALPROTO_FAILED } },
+        _ => { return ALPROTO_UNKNOWN },
+    }
```

Comment:
In a block like this you should be able to remove the `return` statements, this will keep code analyzers happy and is more idiomatic Rust.

Reply:
OK. Thank you!